src/components/EditProfileModal/EditProfileModal.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState , useEffect} from "react";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+import CurrentUserContext from "../../contexts/CurrentUserContext.js";
+import { useForm } from "../../hooks/useForm";
+import { useFormAndValidation } from "../../hooks/useFormAndValidation";
+
+const EditProfileModal = ({
+  handleCloseModal,
+  isOpen,
+  onSubmit,
+  buttonText,
+}) => {
+  const currentUser = React.useContext(CurrentUserContext);
+
+  const formatDate = (dateString) => {
+    const date = new Date(dateString);
+    const year = date.getUTCFullYear();
+    const month = (date.getUTCMonth() + 1).toString().padStart(2, "0");
+    const day = date.getUTCDate().toString().padStart(2, "0");
+    return `${year}-${month}-${day}`;
+  };
+
+  // console.log("currentuser DOB: ", formatDate(currentUser.dob));
+
+  const { values, handleChange, errors, isValid, setValues, setErrors } = useFormAndValidation({
+    name: null,
+    dob: null,
+    placeOfBirth: null,
+    maritalStatus: null,
+    sexualOrientation: null,
+  });
+
+  useEffect(() => {
+    if (isOpen && currentUser) {
+      setValues({
+        name: currentUser.name || "",
+        dob: currentUser.dob ? formatDate(currentUser.dob) : "",
+        placeOfBirth: currentUser.placeOfBirth || "",
+        maritalStatus: currentUser.maritalStatus || "",
+        sexualOrientation: currentUser.sexualOrientation || "",
+      });
+    }
+  }, [isOpen, currentUser, setValues]);
+
+  const handleFormSubmitEdit = (e) => {
+    e.preventDefault();
+    if (isValid) {
+      onSubmit(values);
+    }
+  };
+
+  // console.log("values: ", values);
+  return (
+    <div className="edit__profile">
+      <ModalWithForm
+        title="Change Profile Data"
```

Comment:
* Please, pass `isValid` into the props of `ModalWithForm` to disable the submit button if the form is **not** valid